lua/lspconfig/server_configurations/csharp_ls.lua

```diff
@@ -3,7 +3,12 @@ local util = require 'lspconfig.util'
 return {
   default_config = {
     cmd = { 'csharp-ls' },
-    root_dir = util.root_pattern('*.sln', '*.csproj', '*.fsproj', '.git'),
+    root_dir = function(startpath)
+      return util.root_pattern '*.sln'(startpath)
+        or util.root_pattern '*.csproj'(startpath)
+        or util.root_pattern '*.fsproj'(startpath)
+        or util.root_pattern '.git'(startpath)
+    end,
```

Comment:
why can't the fix be done in `util.root_pattern` instead of special-cased for csharp?

Reply:
Yeah there's no reason why not. I agree that having the special case is weird. Will put something together and we can take a look.